core/core-spacemacs.el

```diff
@@ -249,42 +255,63 @@ defer call using `spacemacs-post-user-config-hook'."
-    ;; Include the backtrace directly in the template, if it exists
-    (setq template (replace-regexp-in-string
-                    "%BACKTRACE%"
-                    (if (get-buffer "*Backtrace*")
-                        (with-current-buffer "*Backtrace*"
-                           (buffer-substring-no-properties
-                            (point-min) (min (point-max) 1000)))
-                      "BACKTRACE IF RELEVANT")
-                    template [keep-case]))
-    ;; Include the last keys description directly into the template, if
-    ;; prefix argument has been passed
-    (setq template (replace-regexp-in-string
-                    "(%LAST_KEYS%)\n"
-                    (if (and arg (y-or-n-p (concat "Do you really want to "
-                                                   "include your last pressed keys? It "
-                                                   "may include some sensitive data.")))
-                        (concat (spacemacs//describe-last-keys-string) "\n")
-                      "")
-                    template [keep-case]))
-    ;; Create the encoded url
-    (setq url (url-encode-url (concat url template)))
+  (let ((buf
+         (generate-new-buffer "REPORT_SPACEMACS_ISSUE"))
+        (system-info
+         (spacemacs//describe-system-info-string))
+        (backtrace
+         (if (get-buffer "*Backtrace*")
+             (with-current-buffer "*Backtrace*"
+               (buffer-substring-no-properties
+                (point-min)
+                (min (point-max) 1000)))
+           "BACKTRACE IF RELEVANT"))
+        (last-keys
+         (if (and arg (y-or-n-p (concat "Do you really want to "
+                                        "include your last pressed keys? It "
+                                        "may include some sensitive data.")))
+             (concat (spacemacs//describe-last-keys-string) "\n")
+           "")))
+    (switch-to-buffer buf)
+    (insert-file-contents-literally
+     (concat configuration-layer-template-directory "REPORTING.template"))
+    (loop
+     for (placeholder replacement)
+     in '(("%SYSTEM_INFO%" system-info)
+          ("%BACKTRACE%" backtrace)
+          ("(%LAST_KEYS%)\n" last-keys))
+     do (save-excursion
+          (goto-char (point-min))
+          (search-forward placeholder)
+          (replace-match (symbol-value replacement) [keep-case] [literal])))
+    (spacemacs/report-issue-mode)))
+
+(define-derived-mode spacemacs/report-issue-mode markdown-mode "Report-Issue"
+  "Major mode for reporting issues with Spacemacs.
+
+When done editing, you can type \\[spacemacs//report-issue-done] to create the
+issue on Github. You must be logged in already for this to work. After you see
+that the issue has been created successfully, you can close this buffer.
+
+Markdown syntax is supported in this buffer.
+
+\\{spacemacs/report-issue-mode-map}
+")
+
+(define-key
+  spacemacs/report-issue-mode-map
+  (kbd "C-c C-c")
```

Comment:
https://github.com/syl20bnr/spacemacs/blob/develop/doc/CONVENTIONS.org#confirm-and-abort